### PR TITLE
DZ06 ready fo checking

### PR DESCRIPTION
**Pull request by igor17011977**
Ready

### CODE REVIEW

File: src/DZ06/ArrayValueCalculator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package DZ06;
+import  java.util.logging.Logger;
+
+import static java.lang.Integer.parseInt;
+
+public class ArrayValueCalculator{
+   int summaArray;
+   int counter;
+    private static final Logger logger = Logger.getLogger(ArrayValueCalculator.class.getName());
+    static class ArraySizeException extends Exception{
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:10]: Usually extends from RuntimeException

Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:10]: You don't need a static class, it crates issues there. A typical separate class is better

File: src/DZ06/ArrayValueCalculator.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        private int a;
+        ArraySizeException(int a){
+            this.a=a;
+        }
+        @Override
+        public String toString(){
+            return ("Array list have "+this.a+" elements");
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:17]: You don't need variable `a`, toString. Please look to super constructor with message field

File: src/DZ06/ArrayValueCalculator.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            this.b=a;
+        }
+        @Override
+        public String toString(){
+            return ("Cell "+this.b+" contains a string value, conversion to int is not possible ");
+        }
+    }
+    public int doCalc(String[][] array) throws ArraySizeException {
+
+        if (array.length > 4) throw new ArraySizeException(array.length);
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:32]: array.length<4?

File: src/DZ06/ArrayValueCalculator.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        for (String[] redo : array
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:33]: not clear variable name `redo`

File: src/DZ06/ArrayValueCalculator.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            ) {
+                for (String sim : redo
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:35]: Your `redo` by task should be also checked by lenght==4

File: src/DZ06/ArrayValueCalculator.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                ) {
+                    try {
+                        counter+=1;
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:38]: You may use for(int i =0; i<redo.lenght; i++). In this is case it more readable

File: src/DZ06/ArrayValueCalculator.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                    int v=parseInt(sim);
+                    this.summaArray+=v;
+                    }catch (NumberFormatException e){System.out.println(new ArrayDataException(counter));}
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:41]: Please don't forget formatting your code

Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:41]: You should throw an exception by your task. There you gen some text

File: src/DZ06/ArrayValueCalculator.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                }
+            }
+        return this.summaArray;
+    }
+    public static void main(String[] args) throws ArraySizeException {
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:46]: By your task you should handle exception in the method `main`

File: src/DZ06/ArrayValueCalculator.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                {"1","s","5","7"},
+                {"1","s","5","7"}};
+        try {
+            String[][] array1 = new String[5][5];
+            new ArrayValueCalculator().doCalc(array1);
+        }catch (ArraySizeException e){
+            logger.warning(""+e);
+        }
+        System.out.println(new ArrayValueCalculator().doCalc(array));
+        //1
```
Comment (SerhiiPasko) [inline src/DZ06/ArrayValueCalculator.java:58]: ?

**Comment by igor17011977** [comment]
close

**Pull request closed without merging**